fhir-server/src/main/java/com/ibm/fhir/server/registry/ServerRegistryResourceProvider.java

```diff
@@ -0,0 +1,215 @@
+                transactionHelper.rollback();
+            }
+        }
+        return Collections.emptyList();
+    }
+
+    private Collection<FHIRRegistryResource> getRegistryResources(Class<? extends Resource> resourceType, Map<String, List<String>> queryParameters) {
+        FHIRTransactionHelper transactionHelper = null;
+        try {
+            FHIRPersistence persistence = persistenceHelper.getFHIRPersistenceImplementation();
+            transactionHelper = new FHIRTransactionHelper(persistence.getTransaction());
+
+            transactionHelper.begin();
+
+            FHIRSearchContext searchContext = SearchUtil.parseQueryParameters(resourceType, queryParameters);
+            searchContext.setPageSize(1000);
+
+            FHIRPersistenceContext context = FHIRPersistenceContextFactory.createPersistenceContext(null, searchContext);
+            MultiResourceResult<Resource> result = persistence.search(context, resourceType);
+
+            if (result.isSuccess()) {
+                List<FHIRRegistryResource> registryResources = new ArrayList<>(searchContext.getTotalCount());
+                registryResources.addAll(result.getResource().stream()
+                        .map(ServerRegistryResource::from)
+                        .filter(Objects::nonNull)
+                        .collect(Collectors.toList()));
+
+                int pageNumber = 1;
+                int lastPageNumber = searchContext.getLastPageNumber();
+                while (pageNumber < lastPageNumber) {
+                    searchContext.setPageNumber(++pageNumber);
+                    result = persistence.search(context, resourceType);
+                    registryResources.addAll(result.getResource().stream()
+                            .map(ServerRegistryResource::from)
+                            .filter(Objects::nonNull)
+                            .collect(Collectors.toList()));
+                }
+
+                transactionHelper.commit();
+                transactionHelper = null;
+
+                return Collections.unmodifiableList(registryResources);
+            }
+        } catch (Exception e) {
+            log.log(Level.WARNING, "An error occurred during a search interaction", e);
+        } finally {
+            if (transactionHelper != null) {
+                transactionHelper.rollback();
+            }
+        }
+
+        return Collections.emptyList();
+    }
+
+    private void updateRegistryResourceMap(FHIRPersistenceEvent event) {
+        if (event == null || event.getFhirResource() == null || !isDefinitionalResource(event.getFhirResource())) {
+            return;
+        }
+        Resource resource = event.getFhirResource();
+        String url = getUrl(resource);
+        if (url != null) {
+            List<FHIRRegistryResource> previous = registryResourceMap.remove(url);
+            if (previous != null && !previous.isEmpty()) {
+                log.info("Removed registry resource(s) with url: " + url);
```

Comment:
Was debating whether INFO level here is appropriate...we try to be very sparse with those.

Reply:
As discussed on slack, please also make it clear we're just removing them from the cache...not from the registry or the db or something

Reply:
Changed to fine